physionet-django/project/models.py

```diff
@@ -683,128 +689,69 @@ def get_anonymous_url(self):
-    is_new_version = models.BooleanField(default=False)
-    # Access url slug, also used as a submitting project id.
-    slug = models.SlugField(max_length=MAX_PROJECT_SLUG_LENGTH, db_index=True)
-    latest_reminder = models.DateTimeField(null=True, blank=True)
-    doi = models.CharField(max_length=50, blank=True, null=True)
-    authors = GenericRelation('project.Author')
-    references = GenericRelation('project.Reference')
-    publications = GenericRelation('project.Publication')
-    topics = GenericRelation('project.Topic')
-
-    authors = GenericRelation('project.Author')
-
-    class Meta:
-        abstract = True
-
-    def __str__(self):
-        return self.title
-
-    def file_root(self):
-        """
-        Root directory containing the project's files
-        """
-        return os.path.join(self.__class__.FILE_ROOT, self.slug)
-
-    def get_storage_info(self, force_calculate=True):
+    def citation_text(self, styles):
         """
-        Return an object containing information about the project's
-        storage usage.
+        Citation information in multiple formats (MLA, APA, Chicago,
+        Harvard, and Vancouver).
 
-        If force_calculate is true, calculate the size by recursively
-        scanning the directory tree.  This is deprecated.
-        """
-        if force_calculate:
-            used = self.storage_used()
-        else:
-            used = None
-        return StorageInfo(allowance=self.core_project.storage_allowance,
-                           used=used, include_remaining=True)
+        1. MLA (8th edition) [https://owl.purdue.edu/owl/research_and_citation/
+                              mla_style/mla_formatting_and_style_guide/
+                              mla_formatting_and_style_guide.html]
+        2. APA (7th edition) [https://owl.purdue.edu/owl/research_and_citation/
+                              apa_style/apa_style_introduction.html]
+        3. Chicago (17th edition) [https://owl.purdue.edu/owl/
+                                   research_and_citation/
+                                   chicago_manual_17th_edition/
+                                   cmos_formatting_and_style_guide/
+                                   chicago_manual_of_style_17th_edition.html]
+        4. Harvard [https://www.mendeley.com/guides/harvard-citation-guide]
+        5. Vancouver [https://guides.lib.monash.edu/ld.php?content_id=14570618]
 
-    def citation_text(self):
-        """
-        Return template citation text.
+        Parameters
+        ----------
+        styles [list]:
+            ['MLA', 'APA', 'Chicago', 'Harvard', 'Vancouver']
 
-        This text resembles the final "citation text" as it will
-        appear once the project is published.  Since the project is
-        not yet published, the current year is used in place of the
-        publication year, and '*****' is used in place of the DOI
-        suffix.
+        Returns
```

Comment:
I think Benjamin suggested keep the returned value of citation_text as a string, rather than a dictionary? If so, perhaps create your dictionary in the view by iterating calls to citation_text() with each individual style that you would like to include?

Reply:
Okay, so have it loop in the view instead? Wasn't sure which would be more useful.

Reply:
Yep, you could then iterate styles in your view to create a dictionary, or alternatively create another method on the model that returns all styles as a dictionary.